From 63be6855389567ba2856eb488ffa927b09879486 Mon Sep 17 00:00:00 2001
From: Khai Wen Ng <khai.wen.ng@intel.com>
Date: Fri, 12 Sep 2025 03:35:20 +0800
Subject: [PATCH 56/61] media: i2c: maxim-serdes: max9296a: update phy and
 stream

Index used for BACKTOP22 to configure PHY and stream ID used for
RX50_STR_SEL is incorrect. Update to use correct value.

Signed-off-by: Khai Wen Ng <khai.wen.ng@intel.com>
---
 drivers/media/i2c/maxim-serdes/max9296a.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/i2c/maxim-serdes/max9296a.c b/drivers/media/i2c/maxim-serdes/max9296a.c
index 9c983cecb6d81..0d9ef4c552a39 100644
--- a/drivers/media/i2c/maxim-serdes/max9296a.c
+++ b/drivers/media/i2c/maxim-serdes/max9296a.c
@@ -549,7 +549,7 @@ static int max9296a_init_phy(struct max_des *des, struct max_des_phy *phy)
 		return ret;
 
 	/* Set DPLL frequency. */
-	ret = regmap_update_bits(priv->regmap, MAX9296A_BACKTOP22(index),
+	ret = regmap_update_bits(priv->regmap, MAX9296A_BACKTOP22(hw_index),
 				 MAX9296A_BACKTOP22_PHY_CSI_TX_DPLL,
 				 FIELD_PREP(MAX9296A_BACKTOP22_PHY_CSI_TX_DPLL,
 					    div_u64(dpll_freq, 100000000)));
@@ -557,7 +557,7 @@ static int max9296a_init_phy(struct max_des *des, struct max_des_phy *phy)
 		return ret;
 
 	/* Enable DPLL frequency. */
-	ret = regmap_set_bits(priv->regmap, MAX9296A_BACKTOP22(index),
+	ret = regmap_set_bits(priv->regmap, MAX9296A_BACKTOP22(hw_index),
 			      MAX9296A_BACKTOP22_PHY_CSI_TX_DPLL_EN);
 	if (ret)
 		return ret;
@@ -727,7 +727,7 @@ static int max9296a_set_pipe_stream_id(struct max_des *des, struct max_des_pipe
 	unsigned int index = max9296a_pipe_id(priv, pipe);
 
 	return regmap_update_bits(priv->regmap, MAX9296A_RX50(index), MAX9296A_RX50_STR_SEL,
-				  FIELD_PREP(MAX9296A_RX50_STR_SEL, pipe->stream_id));
+				  FIELD_PREP(MAX9296A_RX50_STR_SEL, stream_id));
 }
 
 static int max96714_set_pipe_stream_id(struct max_des *des, struct max_des_pipe *pipe,
-- 
2.50.1

